src/snyk/cli/cliExecutable.ts

```diff
@@ -1,37 +1,69 @@
-import * as fs from 'fs/promises';
 import path from 'path';
-import { Platform } from '../common/platform';
-import { Checksum } from './checksum';
+import fs from 'fs/promises';
 import { CliSupportedPlatform } from './supportedPlatforms';
+import { Checksum } from './checksum';
+import { Platform } from '../common/platform';
 
-// TODO: This file is to be removed in VS Code + Language Server feature cleanup. We need to ensure all users have migrated to use CLI path that's set by the language server.
 export class CliExecutable {
-  // If values updated, `.vscodeignore` to be changed.
   public static filenameSuffixes: Record<CliSupportedPlatform, string> = {
     linux: 'snyk-linux',
-    win32: 'snyk-win.exe',
-    darwin: 'snyk-macos',
+    linux_alpine: 'snyk-alpine',
+    macos: 'snyk-macos',
+    macos_arm64: 'snyk-macos-arm64',
+    windows: 'snyk-win.exe',
   };
-
   constructor(public readonly version: string, public readonly checksum: Checksum) {}
 
-  static getFilename(platform: CliSupportedPlatform): string {
-    return this.filenameSuffixes[platform];
-  }
-
-  static getPath(extensionDir: string, customPath?: string): string {
+  static async getPath(extensionDir: string, customPath?: string): Promise<string> {
     if (customPath) {
       return customPath;
     }
 
-    const platform = Platform.getCurrent();
-    const fileName = CliExecutable.getFilename(platform as CliSupportedPlatform);
+    const platform = await this.getCurrentWithArch();
+    const fileName = this.getFileName(platform);
     return path.join(extensionDir, fileName);
   }
 
-  static exists(extensionDir: string, customPath?: string): Promise<boolean> {
+  static getFileName(platform: CliSupportedPlatform): string {
+    return this.filenameSuffixes[platform];
+  }
+
+  static async getCurrentWithArch(): Promise<CliSupportedPlatform> {
+    let platform = '';
+    const osName = Platform.getCurrent().toString().toLowerCase();
+    const archName = Platform.getArch().toLowerCase();
+    if (osName === 'linux') {
+      if (await this.isAlpine()) {
+        platform = 'linux_alpine';
+      } else {
+        platform = 'linux';
+      }
+    } else if (osName === 'darwin') {
+      if (archName === 'arm64') {
+        platform = 'macos_arm64';
```

Comment:
same here - would think it needs to be a dash. can we test that in integration tests on the different operating systems?